hydrolib/tools/extforce_convert/utils.py

```diff
@@ -200,8 +201,50 @@ def convert_interpolation_data(
     return data
 
 
+def path_relative_to_parent(
+    forcing: ExtOldForcing,
+    inifile_path: Path,
+    ext_old_path: Path,
+    mdu_parser: Any,
+) -> Path:
+    """Resolve the path of the forcing file relative to the parent file if needed.
+
+    Args:
+        forcing (ExtOldForcing):
+            The old forcing block with the filename to resolve.
+        inifile_path (Path):
+            The path to the inifields file.
+        ext_old_path (Path):
+            The path to the old external forcing file.
+        resolve_parent (bool):
+            Whether to resolve the path relative to the parent file.
+            if True, the path is resolved relative to the parent file.
+            if False, the path is returned as is because it is relative to the mdu file.
+
+    Returns:
+        Path: The resolved path of the forcing file.
+    """
+    if mdu_parser is None:
+        resolve_parent = False
+    else:
+        general_mdu = mdu_parser.loaded_fm_data.get("general", {})
+        resolve_parent = general_mdu.get("pathsrelativetoparent", "0") == "1"
+    update_path_condition = (
+        forcing.filename.filepath.is_absolute() or not resolve_parent
+    )
+    forcing_path = (
+        forcing.filename.filepath
+        if update_path_condition
+        else os.path.relpath(
+            ext_old_path.parent / forcing.filename.filepath, inifile_path.parent
+        )
+    )
+    return forcing_path
+
+
 def create_initial_cond_and_parameter_input_dict(
     forcing: ExtOldForcing,
+    forcing_path: Path,
```

Comment:
why do you have to path the forcing path here? when you can reassign the forcing path to the `forcing.filename.filepath` outside the function and and then no changes are needed here ?